lib/storage/src/content_manager/consensus_state.rs

```diff
@@ -323,6 +324,15 @@ impl<C: CollectionContainer> ConsensusState<C> {
         self.persistent.write().insert_peer(peer_id, uri)
     }
 
+    pub fn remove_peer(&self, peer_id: PeerId) -> Result<(), StorageError> {
+        if self.toc.peer_has_shards(peer_id) {
```

Comment:
Should we check if the peer is currently receiving a shard via a transfer?

Reply:
Yes, good point. I guess I will add it in the next PR then :smile: 

Also because of this discussion I noticed that we capture proxy shards in collection snapshot, which is interesting. I will need to recheck the logic here.